src/components/Web3Provider/index.tsx

```diff
@@ -0,0 +1,137 @@
+import { useWeb3React, Web3ReactProvider } from '@web3-react/core'
+import { coinbaseWallet, gnosisSafe, injected, walletConnect } from 'connectors'
+import { connectors, network } from 'connectors'
+import { getConnectorForWallet, Wallet } from 'constants/wallet'
+import usePrevious from 'hooks/usePrevious'
+import { useEffect, useState } from 'react'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { updateWalletOverride } from 'state/user/reducer'
+
+const GNOSIS_SAFE_URL = 'https://gnosis-safe.io'
+const IS_GNOSIS_SAFE = window.location.ancestorOrigins.contains(GNOSIS_SAFE_URL)
+
+interface ConnectorState {
+  isActive: boolean
+  previousIsActive: boolean | undefined
+  isActivating: boolean
+  isEagerlyConnecting: boolean
+  setIsEagerlyConnecting(connecting: boolean): void
+}
+
+const Web3Updater = () => {
```

Comment:
Add a doc comment explaining what `Web3Updater` does (because it's not obvious and hard to read through the logic).